13_import.sql

```diff
@@ -0,0 +1,575 @@
+  END IF;
+
+  -- set structure okay
+  UPDATE qgep_import.manhole_quarantine
+  SET structure_okay = true
+  WHERE quarantine_serial = NEW.quarantine_serial;
+  RETURN NEW;
+
+  -- catch
+  EXCEPTION WHEN OTHERS THEN
+    RAISE NOTICE 'EXCEPTION: %', SQLERRM;
+    RETURN NEW;
+END; $BODY$
+LANGUAGE plpgsql;
+
+DROP TRIGGER IF EXISTS after_update_try_structure_update ON qgep_import.manhole_quarantine;
+
+CREATE TRIGGER after_update_try_structure_update
+  AFTER UPDATE
+  ON qgep_import.manhole_quarantine
+  FOR EACH ROW
+  WHEN ( ( NEW.structure_okay IS NOT TRUE ) 
+  AND NOT( OLD.inlet_okay IS NOT TRUE AND NEW.inlet_okay IS TRUE )
+  AND NOT( OLD.outlet_okay IS NOT TRUE AND NEW.outlet_okay IS TRUE ) )
+  EXECUTE PROCEDURE qgep_import.manhole_quarantine_try_structure_update();
+
+DROP TRIGGER IF EXISTS after_insert_try_structure_update ON qgep_import.manhole_quarantine;
+
+CREATE TRIGGER after_insert_try_structure_update
+  AFTER INSERT
+  ON qgep_import.manhole_quarantine
+  FOR EACH ROW
+  EXECUTE PROCEDURE qgep_import.manhole_quarantine_try_structure_update();
+
+
+-- Some information:
+-- 1. new lets 0 - old lets 0 -> do nothing
+-- 2. new lets 0 - old lets 1 -> delete let
+-- 3. new lets 0 - old lets n -> delete lets
+-- 4. new lets 1 - old lets 0 -> create let (future)
+-- 5. new lets 1 - old lets 1 -> update let
+-- 6. new lets 1 - old lets n -> manual update needed
+-- 7. new lets n - old lets 0 -> create lets (future)
+-- 8. new lets n - old lets 1 -> manual update needed
+-- 9. new lets n - old lets n -> manual update needed
+
+CREATE OR REPLACE FUNCTION qgep_import.manhole_quarantine_try_inlet_update() RETURNS trigger AS $BODY$
+  DECLARE 
+    new_lets integer;
+    old_lets integer;
+BEGIN
+  -- count new lets
+  IF NEW.inlet_3_material IS NOT NULL OR NEW.inlet_3_depth_m IS NOT NULL OR NEW.inlet_3_clear_hight IS NOT NULL THEN
+    IF NEW.inlet_4_material IS NOT NULL OR NEW.inlet_4_depth_m IS NOT NULL OR NEW.inlet_4_clear_hight IS NOT NULL THEN
+      new_lets = 2; -- it's possibly more, but at least > 1
+    ELSE
+      new_lets = 1;
+    END IF;
+  ELSE
+    new_lets = 0;
+  END IF;
+  -- count old lets
+  old_lets = ( SELECT COUNT (*)
+    FROM qgep_od.reach re
+    LEFT JOIN qgep_od.reach_point rp ON rp.obj_id = re.fk_reach_point_to
+    LEFT JOIN qgep_od.wastewater_networkelement wn ON wn.obj_id = rp.fk_wastewater_networkelement
+    LEFT JOIN qgep_od.vw_qgep_wastewater_structure ws ON ws.obj_id = wn.fk_wastewater_structure
+    WHERE ws.obj_id = NEW.obj_id );
+
+  -- handle inlets
+  IF ( new_lets > 1 AND old_lets > 0 ) OR old_lets > 1 THEN
+    -- request for update because new lets are bigger 1 (and old lets not 0 ) or old lets are bigger 1
+    RAISE NOTICE 'Impossible to assign inlets - manual edit needed.';
+  ELSE
+    IF new_lets = 0 AND old_lets > 0 THEN
+      -- request for delete because no new lets but old lets
+      RAISE NOTICE 'No new inlets but old ones - manual delete needed.';
```

Comment:
shall this be translated?

Reply:
I don't know, but if we have the possibility this would be nice. Is there anything existing already regarding that?

Reply:
the only thing I would think of is to create a table `qgep_sys.error_messages` which has the translation, a setting for the lang (probably another table `qgep_sys.settings` (see [qwat](https://github.com/qwat/qwat-data-model/blob/master/system/settings.sql) for instance).
but this is maybe too much for here.
ping @m-kuhn

Reply:
Still open. Do I need to do something before merging?

Reply:
I'm not sure we want this translation on database level. Might be something to integrate in the QGIS translation. Not a requirement for merging this.